### ISSUE TITLE
Keycloak stoppes after first import

### ISSUE THREAD

**Issue by yelhouti** [keycloak/keycloak#10322]
### Describe the bug

When I start keycloak with: `'-Dkeycloak.migration.dir=/opt/jboss/keycloak/realm-config'`

It import successfully then gracefully stops for some reason.
here are the logs:
> keycloak_1  | Updating the configuration and installing your custom providers, if any. Please wait.
keycloak_1  | 2022-02-18 17:21:15,205 INFO  [org.keycloak.common.Profile] (build-14) Preview feature enabled: scripts
keycloak_1  | 2022-02-18 17:21:20,448 INFO  [io.quarkus.deployment.QuarkusAugmentor] (main) Quarkus augmentation completed in 6143ms
keycloak_1  | 2022-02-18 17:21:22,680 INFO  [org.keycloak.common.Profile] (main) Preview feature enabled: scripts
keycloak_1  | 2022-02-18 17:21:22,743 INFO  [org.keycloak.quarkus.runtime.hostname.DefaultHostnameProvider] (main) Hostname settings: FrontEnd: <request>, Strict HTTPS: false, Path: <request>, Strict BackChannel: false, Admin: <request>
keycloak_1  | 2022-02-18 17:21:23,159 WARN  [org.infinispan.PERSISTENCE] (keycloak-cache-init) ISPN000554: jboss-marshalling is deprecated and planned for removal
keycloak_1  | 2022-02-18 17:21:23,240 WARN  [org.infinispan.CONFIG] (keycloak-cache-init) ISPN000569: Unable to persist Infinispan internal caches as no global state enabled
keycloak_1  | 2022-02-18 17:21:23,289 INFO  [org.infinispan.CONTAINER] (keycloak-cache-init) ISPN000556: Starting user marshaller 'org.infinispan.jboss.marshalling.core.JBossUserMarshaller'
keycloak_1  | 2022-02-18 17:21:23,468 INFO  [org.infinispan.CONTAINER] (keycloak-cache-init) ISPN000128: Infinispan version: Infinispan 'Triskaidekaphobia' 13.0.5.Final
keycloak_1  | 2022-02-18 17:21:23,818 INFO  [org.keycloak.connections.infinispan.DefaultInfinispanConnectionProviderFactory] (main) Node name: node_416038, Site name: null
keycloak_1  | 2022-02-18 17:21:24,812 INFO  [org.keycloak.quarkus.runtime.storage.database.liquibase.QuarkusJpaUpdaterProvider] (main) Initializing database schema. Using changelog META-INF/jpa-changelog-master.xml
keycloak_1  | 2022-02-18 17:21:26,669 INFO  [org.keycloak.exportimport.dir.DirImportProvider] (main) Importing from directory /opt/jboss/keycloak/realm-config
keycloak_1  | 2022-02-18 17:21:26,774 INFO  [org.keycloak.services] (main) KC-SERVICES0050: Initializing master realm
keycloak_1  | 2022-02-18 17:21:28,078 INFO  [org.keycloak.services] (main) KC-SERVICES0030: Full model import requested. Strategy: OVERWRITE_EXISTING
keycloak_1  | 2022-02-18 17:21:30,054 INFO  [org.keycloak.exportimport.util.ImportUtils] (main) Realm 'jhipster' imported
keycloak_1  | 2022-02-18 17:21:30,237 INFO  [org.keycloak.exportimport.dir.DirImportProvider] (main) Imported users from /opt/jboss/keycloak/realm-config/jhipster-users-0.json
keycloak_1  | 2022-02-18 17:21:30,279 INFO  [org.keycloak.services] (main) KC-SERVICES0032: Import finished successfully
keycloak_1  | 2022-02-18 17:21:30,418 INFO  [org.keycloak.services] (main) KC-SERVICES0009: Added user 'admin' to realm 'master'
keycloak_1  | 2022-02-18 17:21:30,512 INFO  [io.quarkus] (main) Keycloak 17.0.0 on JVM (powered by Quarkus 2.7.0.Final) started in 9.898s. Listening on: http://0.0.0.0:8080
keycloak_1  | 2022-02-18 17:21:30,513 INFO  [io.quarkus] (main) Profile dev activated. 
keycloak_1  | 2022-02-18 17:21:30,513 INFO  [io.quarkus] (main) Installed features: [agroal, cdi, hibernate-orm, infinispan-client, jdbc-h2, jdbc-mariadb, jdbc-mssql, jdbc-mysql, jdbc-oracle, jdbc-postgresql, keycloak, narayana-jta, reactive-routes, resteasy, resteasy-jackson, smallrye-context-propagation, smallrye-health, smallrye-metrics, vault, vertx]
keycloak_1  | 2022-02-18 17:21:30,514 WARN  [org.keycloak.quarkus.runtime.KeycloakMain] (main) Running the server in development mode. DO NOT use this configuration in production.
keycloak_1  | 2022-02-18 17:21:30,752 INFO  [io.quarkus] (main) Keycloak stopped in 0.237s
docker_keycloak_1 exited with code 0

If I restart the container it works with no issue.

### Version

17.0.0

### Expected behavior

To start and keep running the first time

### Actual behavior

It stops after import

### How to Reproduce?

docker-compose.yaml
```yaml
# This configuration is intended for development purpose, it's **your** responsibility to harden it for production
version: '3.8'
services:
  keycloak:
    image: quay.io/keycloak/keycloak:17.0.0
    command:
      [
        'start-dev',
        '-Dkeycloak.migration.action=import',
        '-Dkeycloak.migration.provider=dir',
        '-Dkeycloak.migration.dir=/opt/jboss/keycloak/realm-config',
        '-Dkeycloak.migration.strategy=OVERWRITE_EXISTING',
        '-Dkeycloak.profile.feature.scripts=enabled',
      ]
    volumes:
      - ./realm-config:/opt/jboss/keycloak/realm-config
    environment:
      - KEYCLOAK_ADMIN=admin
      - KEYCLOAK_ADMIN_PASSWORD=admin

```

**Comment by pedroigor** [comment]
@yelhouti, could you please check https://github.com/keycloak/keycloak/pull/10754?

**Comment by yelhouti** [comment]
Seems to be exactly what I want. What do you want me to do ?

**Comment by MetroMarv** [comment]
If you build your own Keycloak docker image, you can also workaround this bug by creating a new script looks as follows:

```shell
#!/bin/sh

/opt/keycloak/bin/kc.sh start-dev \
  -Dkeycloak.migration.action=import \
  -Dkeycloak.migration.provider=dir \
  -Dkeycloak.migration.dir=<Your directory to import> \
  -Dkeycloak.migration.strategy=<Your Strategy>

/opt/keycloak/bin/kc.sh start-dev
```

It will simply start another Keycloak that will not shutdown.

**Comment by stianst** [comment]
Closing as this is resolved by #9261
